src/status_im/chat/events/input.cljs

```diff
@@ -237,77 +237,50 @@
-  (let [params-template      {:content      content
-                              :chat-id      current-chat-id
-                              :group-id     (when (get-in chats [current-chat-id :group-chat])
-                                              current-chat-id)
-                              :jail-id      (or bot current-chat-id)
-                              :message-id   message-id
-                              :current-time current-time} 
-        preview-params       (merge params-template
-                                    {:data-type           :preview
-                                     :event-after-creator (fn [command-message jail-response]
-                                                            [::send-command
-                                                             (assoc-in command-message [:command :preview] jail-response)])})
-        short-preview-params (merge params-template
-                                    {:data-type           :short-preview
-                                     :event-after-creator (fn [_ jail-response]
-                                                            [::request-command-data
-                                                             (assoc-in preview-params [:content :command :short-preview] jail-response)])})
-        validation-params    (merge params-template
-                                    {:data-type           :validator
-                                     :event-after-creator (fn [_ jail-response]
-                                                            [::proceed-validation
-                                                             jail-response
-                                                             [[::request-command-data short-preview-params]]])})]
-    (request-command-data db validation-params)))
+  [{:keys [bot-db current-chat-id chats] :as db} {:keys [command metadata] :as command-params} message-id current-time]
+  (let [message     (command->message db command-params)
+        cmd-params  {:command    command
+                     :params     (get-in message [:content :params])
+                     :to-message (:to-message-id metadata)
+                     :created-at current-time
+                     :id         message-id
+                     :chat-id    current-chat-id
+                     :jail-id    (:jail-id message)}
+        event-chain {:data-type             :validator
+                     :proceed-event-creator (fn [validation-response]
+                                              [::proceed-validation
+                                               validation-response
+                                               [[:request-command-message-data
+                                                 message
+                                                 {:data-type             :short-preview
+                                                  :proceed-event-creator (fn [short-preview]
```

Comment:
could this be turned into a function to reduce nesting or is it not worth ?

Reply:
I thought about it, but I think it's actually more obvious and easier to follow what's happening when it's expressed this way.

Reply:
I vote for separate functions because it's hard to read